api/models/usersDirectMessageThreads.js

```diff
@@ -92,6 +92,49 @@ const updateDirectMessageThreadNotificationStatusForUser = (threadId: string, us
 ===========================================================
 */
 
+const archiveDirectMessageThread = (
+  userDirectMessageThreadId: string
+): Promise<Object> => {
+  return db
+    .table('usersDirectMessageThreads')
+    .get(userDirectMessageThreadId)
+    .update(
+      {
+        archivedAt: new Date(),
+        receiveNotifications: false,
+      },
+      { returnChanges: true }
+    )
+    .run();
+};
+
+const unarchiveDirectMessageThread = (
+  userDirectMessageThreadId: string
+): Promise<Object> => {
+  return db
+    .table('usersDirectMessageThreads')
+    .get(userDirectMessageThreadId)
+    .update(
+      {
+        archivedAt: db.literal(),
+        receiveNotifications: true,
```

Comment:
This should **turn notifications on** when a thread is unarchived, right? @mxstbr @brianlovin

Reply:
I don't think archiving and unarchiving should affect the notification settings - all it should do is hide it from the UI. New messages in that thread should definitely send notifications and mark it as un-archived right?

Reply:
@brianlovin in the initial requirements we had:

> Archiving direct message threads: Let users archive DM threads; archived threads are hidden/not loaded for the user who archived it(!) when navigating to the main DM screen; no notifications are sent for new messages in archived threads; there's some sort of tab or filter that let's users look at their archived threads and un-archive them

As we discussed previously, we say that the only way that a message comes back to active and start sending notifications again is when you:
a) send a message in the archive thread
b) click "unarchive"

Am I missing something? I know rules can change pretty quickly though :P

Reply:
Ah sorry, yeah I think we miscommunicated there or we had inconsistent communication - that's my bad. I think that there is a difference between muting a thread and archiving a thread. Muting would imply that you no longer want to receive notifications. Archiving is more about inbox management and clearing out old conversations to make the direct messages list easier to parse. But a new message in an archived thread is still important - so we resurface it and send a new notification.

Reply:
If I understand this correctly is that if I want to `mute` a thread I should update  `receiveNotifications` to false and vice-versa. Correct?

Reply:
And muting a thread will be another task.